src/data-access/repositories/mongo/mongo-device-repository.ts

```diff
@@ -1,24 +1,56 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
 import { BaseMongoRepository } from './base-mongo-repository'
 import { DeviceRepository } from '../interfaces/device-repository'
 import { Device } from '../../../model/entities/device'
 import { MongoDatabase } from './mongo-database'
-import { MongoDevice, MongoEntityConverter } from './mongo-entity-converter'
+import { INewsDevice } from '../../../model/entities/inews-device'
+import { TelemetricsDevice } from '../../../model/entities/telemetrics-device'
 
-const DEVICE_COLLECTION_NAME: string = 'peripheralDevices'
+const DEVICE_COLLECTION_NAME: string = 'externalDevices'
 
 export class MongoDeviceRepository extends BaseMongoRepository implements DeviceRepository {
 
-  constructor(mongoDatabase: MongoDatabase, private readonly mongoEntityConverter: MongoEntityConverter) {
+  constructor(mongoDatabase: MongoDatabase) {
     super(mongoDatabase)
   }
 
   protected getCollectionName(): string {
     return DEVICE_COLLECTION_NAME
   }
 
-  public async getDevices(): Promise<Device[]> {
-    this.assertDatabaseConnection(MongoDeviceRepository.prototype.getDevices.name)
-    const mongoDevices: MongoDevice[] = await this.getCollection().find<MongoDevice>({}).toArray()
-    return this.mongoEntityConverter.convertToDevices(mongoDevices)
+  public async findAllDevices(): Promise<(Device | INewsDevice | TelemetricsDevice)[]> {
+    this.assertDatabaseConnection(MongoDeviceRepository.prototype.findAllDevices.name)
+    const devices: Device[] = await this.getCollection().find<(Device | INewsDevice | TelemetricsDevice)>({}).toArray() 
+    return devices
+  }
+
+  public async findById(_deviceId: string): Promise<Device | null> {
+    this.assertDatabaseConnection(MongoDeviceRepository.prototype.findById.name)
+    try {
+      const device = await this.getCollection().findOne<(Device | INewsDevice | TelemetricsDevice)>({ _id: _deviceId })
+      return device
+    } catch (error) {
+      console.error(`Error finding device by id ${_deviceId}:`, error)
+      return null
+    }
+  }
```

Comment:
We don't want to return `null`. If you need to return a value that's "not there" then use `undefined`. `null` and `undefined` behaves differently.

But, we do prefer to never return `undefined` as well, when we can get away with it.
What we have done in the other repositories is something like this:
````
public async findById(deviceId: string): Promise<Device> {
  // assertion here
  const device: Device | null = await this.getCollection().findOne<Device}({ _id: deviceId })
  if (!device) {
    throw new NotFoundException(...)
  }
  return device
}
```